kapitan/targets.py

```diff
@@ -324,6 +340,46 @@ def compile_target(target_obj, search_paths, compile_path, ref_controller, **kwa
     logger.info("Compiled %s (%.2fs)", target_name, time.time() - start)
 
 
+def get_dependencies_per_target(target_objs):
+    """returns a mapping from target name to a list of dependency objects"""
+    mapping = {}
+    for target_obj in target_objs:
+        try:
+            dependencies = target_obj["dependencies"]
+            target_name = target_obj["vars"]["target"]
+            mapping[target_name] = []
+            for item in dependencies:
+                dependency_type = item["type"]
+                source_uri = item["source"]
+                output_path = item["output_path"]
+                if dependency_type == "git":
+                    del item["output_path"]
+                    mapping[target_name].append(Git(source_uri, output_path, **item))
+                elif dependency_type in ("http", "https"):
+                    mapping[target_name].append(Http(source_uri, output_path))
+                else:
+                    logger.error("type {} is not supported as dependency".format(dependency_type))
+                    assert False
+
+        except KeyError:
+            continue
+    return mapping
+
+
+def remove_saved_dependencies(dependency_mapping, cache_path=None):
+    if cache_path is not None:
+        Dependency.set_cache_path(cache_path)
```

Comment:
Is this going to work correctly when compiling in parallel? I'm worried about race conditions

Reply:
@ramaro @uberspot 
I'm assuming the user wouldn't specify the same dependency source twice for each target as it doesn't make sense to save the same file in two different output_paths. In that case there wouldn't be any race condition.
Maybe we need to redesign the code for fetching? Since fetched resources are not per-target, it doesn't make sense to follow how `ref`s are written (i.e. Ref object/class for each secret). It is probably easier to loop through the `dependencies` parameter across targets, remove duplicate sources to make the sources unique, fetch and save to tmp, and then copy those into their respective output_path(s) if no file at those path(s) exists. In my opinion, having separate Git/Http instances makes the fetching process rather convoluted.

Reply:
@yoshi-1224  Should this be marked as Resolved since it was revised in a newer commit in this PR?